src/components/MetadataDialog/useMetadataForm.ts

```diff
@@ -0,0 +1,136 @@
+import { MetadataFormData } from "@dashboard/components/Metadata";
+import { EventDataAction, EventDataField } from "@dashboard/components/Metadata/types";
+import { getDataKey, parseEventData } from "@dashboard/components/Metadata/utils";
+import { MetadataInput } from "@dashboard/graphql";
+import { ChangeEvent } from "@dashboard/hooks/useForm";
+import { flattenErrors } from "@dashboard/utils/hook-form/errors";
+import { mapMetadataItemToInput } from "@dashboard/utils/maps";
+import { useMemo } from "react";
+import { FieldError, useFieldArray, useForm, UseFormReset } from "react-hook-form";
+import { useIntl } from "react-intl";
+
+import { getValidateMetadata } from "./validation";
+
+interface UseMetadataFormConfig {
+  entityData: { metadata: any; privateMetadata: any } | undefined;
```

Comment:
metadata is not `any`. Maybe let's call it `EntityWithMetadata`, you can also do 

```
interface UseMetadataFormConfig<EntityWithMetadata extends {metadata: MetadataItem[]; privateMetadata: MetadataItem[]}>
```

or without generic, but typed. And rename (entityData is not best name)